src/debuggability.js

```diff
@@ -128,6 +130,68 @@ Promise.hasLongStackTraces = function () {
     return config.longStackTraces && longStackTracesIsSupported();
 };
 
+function registerPromise() {
+    if (Promise.monitor) {
+        Promise.monitor._promiseIdCounter++;
+        this._promiseId = Promise.monitor._promiseIdCounter;
+        Promise.monitor._pendingPromises[Promise.monitor._promiseIdCounter]
+            = this;
+    }
+}
+
+function unregisterPromise() {
+    if (Promise.monitor)
+        delete Promise.monitor._pendingPromises[this._promiseId];
```

Comment:
This probably  doesn't need to be accessible from the promise object? It could just be a private closure variable